climada/hazard/centroids/centr.py

```diff
@@ -906,23 +906,36 @@ def write_hdf5(self, file_name, mode="w"):
             (path and) file name to write to.
         """
         LOGGER.info("Writing %s", file_name)
-        store = pd.HDFStore(file_name, mode=mode)
-        pandas_df = pd.DataFrame(self.gdf)
-        for col in pandas_df.columns:
-            if str(pandas_df[col].dtype) == "geometry":
-                pandas_df[col] = np.asarray(self.gdf[col])
-
-        # Avoid pandas PerformanceWarning when writing HDF5 data
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore", category=pd.errors.PerformanceWarning)
-            # Write dataframe
-            store.put("centroids", pandas_df)
-
-        store.get_storer("centroids").attrs.metadata = {
-            "crs": CRS.from_user_input(self.crs).to_wkt()
-        }
-
-        store.close()
+        xycols = []
+        wkbcols = []
+        store = pd.HDFStore(file_name, mode=mode, complevel=9)
+        try:
+            pandas_df = pd.DataFrame(self.gdf)
+            for col in pandas_df.columns:
+                if str(pandas_df[col].dtype) == "geometry":
```

Comment:
Make clear that you do not want to iterate over all columns:
```suggestion
            for col in filter(lambda x: str(x.dtype) == "geometry", pandas_df.columns):
```
(Suggestion won't work because the following code needs to be indented less)

Reply:
elegant suggestion - but I leave it as it is. it's more "climada style" like that.